### ISSUE TITLE
"System" submenu under Management item

### ISSUE THREAD

**Issue by edgarf** [EASOL/easol#156]
We need one more sub-menu item in Management section, it should be called "System".

The page would have the following sections
- [x] Data Sceme Update
- [x] Time Zone Selection (iceboxed yet)
- [x] Module Management
- [x] Google Auth Management
- [x] Langauge Selection

#### Data Schema Update
Since all fo the changes below will be applied generally for the system and also we would like to use our current "key,value" structured table ```EASOL.SchoolConfiguration```, some database schema and code changes would need to be done.
1) Currently, this table can't accept null as a value. However, we would like to provide a ```null``` value for the non specific school related options. 
2) Change Table title to "EASOL.Configuration" because this will represent not only School level, but also the system level configurations
3) Check all current EASOL Dashboard code and change it in required places to fwork with our 

#### Time Zone Selection
Some parts of the system rely a lot on the time-zone. (For example Analytics API needs time when the sections are taking place) that's why the deviation of even 1 hour can distort the data coming from our API).
The selected time zone can be saved in ```EASOL.Configuration``` and then the required code changes done to make sure the 

#### Module Management
We need to allow for plug-in modules to the EASOL Dashboard. This will allow us in the future to add more third-party modules easily and also currently - manage current Content and Analytics service usage.
Right now we have these two services integrated:
a) Learning Tapestry Content API Search (```/content```)
b) Learning Tapestry Analytics API (```/analytics```)
These two need to be converted into a library (plugin).

When this is done, then we need to add a trigger to Enable/Disable plugins (libraries) from the "System" submenu in "Management". Additionally
 a) Analytics - will need to have API key, API pass and API URL values
 b) Content - we will need only the Content API url for now.

#### Google Auth Management
Currently system allows for users to login using their Google Accounts. "System" submenu should offer to enable/disable it and also provide ability to enter Google App identificator. 

When enabling the plugin, script should check whether the database has the field for these purposes - it's ```GoogleAuth``` field in ```easol.StaffAuthentication``` table.

Related parts of the functionality which should be adjusted to check if Google Auth is enabled:
a) Login button on the login view
b) Authorization process
c) User Creation Process (whether to allow users to login with their emails or Google Auth).

#### Language Selection

This functionality will be introduced later, just add a section called "Language Selection" on the page.

**Comment by regiscamimura** [comment]
I have it done and pushed to 156-feature-system-submenu branch.

**Comment by regiscamimura** [comment]
Previous discussion on timezone issues:

>     General question - could you please tell me how is the Timezone setting affecting the rest of the system? Analytics is the most important part for us. Do you convert the received values from analytics? The time values in analytics are in UTC.
>     I've put a function there that uses a native PHP function which sets the timezone globally (using "date_default_timezone_set()" function). But there are global concerns here. We should:
>     - Every time we display a date, that should go through a custom helper. I've already put such function (in application/helpers/datetime_helper.php), but we need to go through the views and make sure we're using it. By doing that we make the code cleaner, we can define a global date formatting, and if we need to change anything related to dates, we change only this helper function instead of going through all controllers and views.
>     - There is also a concern related to writing of dates to the DB. Should we store dates there using PHP timezone, or allow the dates in the DB to be in a different timezone?
>      
>     Anyway, I'm not really sure if the function I put is getting us the expected results. I couldn't really see where date/times are being displayed or retrieved in the analytics section. Could you please provide me some simple cases of usage and expected results?
	Thank you for the explanation. We would be very thankful if you could go through the views and convert all date to be represented with your helper. 
	1) The place where the dates are used in the system where some time zones problem can happen is in Analytics. Firstly the system gets the time intervals(a), when the periods are happening, those dates are taking 			from edfi.* database (Those are in the timezone we select in the system tab) and then those intervals are sent to the Analytics API(b) (those should be converted into UTC timezone) and then the Analytics API is 			sending back the data(c) with UTC dates, which should be converted into the “local” timezone. The exact places in code are here:
		(a) = application/modules/analytics/controllers/Analytics.php lines 92, 219, 332
		(b) = application/modules/analytics/controllers/Analytics.php lines 140,252, 355
		(c) = application/modules/analytics/controllers/Analytics.php lines 145, 255, 360
	2) Regarding storage to the database, I don’t think we are now having any kind of time/date storage to the database - let me know if you saw any. The only ones I can think of are columns “Date Created”/“Date Modified” kind of ones.

**Comment by regiscamimura** [comment]
I've created a function to pass forward the time in UTC timezone to the API. But regarding the values returned by the API, we don't actually have times in a day, but how many hours/minutes a user spent in a section, so if I got it right, it doesn't makes sense to convert the values returned by the API from UTC to the system's timezone. Am I missing anything? 

I also modified the views so the date goes through the date helper.
